andela/ah-backend-prime
# #164798301 Users can tag their articles

**What does this PR do?**
   Creates functionality for users to be able to tag their articles

**Description of tasks to be completed**

- Create the ability for users to add tags when creating articles

- Create functionality for fetching all available tags

**How should this be manually tested?**

- After cloning the repo, cd into it and run the following commands to activate the virtual environment
   `virtualenv venv`
   `source venv/bin/activate`

- Checkout the branch with the tags functionality
  ` git checkout ft-tag-articles-164798301`

- Install all requirements
   `pip install -r requirements.txt`

- Source the environment variables
   `source .env`

- Run the migrations
   `python manage.py makemigrations`
   `python manage.py migrate`

- Run the server
   `python manage.py runserver`

- To test out this functionality one has to first register on the application, in case of a successful 
   registration, they will receive an account activation link on their email, this activation link would then 
   be pasted in the URL section of postman and sent in order to activate and log into one's account 
   where they would get a token which they would use in the Authorization section to access all other 
   routes including the tags route.

- To add a tag to an article, you should simply add a tags field to the article. This is a list that has to 
   contain all the tags you are adding to the article. Then send a POST request to the url 
  ` http://127.0.0.1:8000/api/v1/articles/`.  (see Figure 1)
   
<img width="1216" alt="Screen Shot 2019-04-14 at 10 46 25" src="https://user-images.githubusercontent.com/26630001/56089895-b1127300-5ea2-11e9-8d2a-2af8b87c2990.png">

**Figure 1**

- The result should be:

<img width="1210" alt="Screen Shot 2019-04-14 at 10 46 52" src="https://user-images.githubusercontent.com/26630001/56089907-e4550200-5ea2-11e9-9eac-31d7935146dd.png">

- To fetch all available tags, you should send a GET request to the url `http://127.0.0.1:8000/api/v1/tags/`. (See Figure 2)

<img width="1209" alt="Screen Shot 2019-04-14 at 10 51 54" src="https://user-images.githubusercontent.com/26630001/56089958-5e858680-5ea3-11e9-80e2-3b7148733f16.png">

**Figure 2**

- **What are the relevant pivotal tracker stories?**
   [#164798301](https://www.pivotaltracker.com/story/show/164798301)
